rancher-reaper/README.md

```diff
@@ -0,0 +1,119 @@
+
+#Running the Service
+You will generally run one instance of this container in each Rancher environment.The following Rancher config should provide you with a good starting point:
+
+docker-compose.yml:
+
+rancher-reaper:
+  image: ampedandwired/rancher-reaper:latest
+  tty: true
+  environment:
+    AWS_ACCESS_KEY_ID: ${AccessKeyId}
+    AWS_SECRET_ACCESS_KEY: ${SecretAccessKey}
+  labels:
+    io.rancher.container.create_agent: 'true'
+    io.rancher.container.agent.role: environment
+rancher-compose.yml:
+
+rancher-reaper:
+  scale: 1
+  health_check:
+    port: 3000
+    interval: 2000
+    unhealthy_threshold: 3
+    strategy: recreate
+    response_timeout: 2000
+    request_line: GET / HTTP/1.0
+    healthy_threshold: 2
+This container requires the following environment variables to be set:
+
+CATTLE_URL
+CATTLE_ACCESS_KEY
+CATTLE_SECRET_KEY
+AWS_ACCESS_KEY_ID
+AWS_SECRET_ACCESS_KEY
+The easiest way to set the CATTLE_* variables is to set up your container with a service account by applying the following labels:
+
+io.rancher.container.create_agent: true
+io.rancher.container.agent.role: environment
+The AWS_* variables should contain AWS API keys that have ec2:DescribeInstances and ec2:DescribeRegions permissions on all resources.
+Use this IAM policy as a guide:
+
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "ec2:DescribeInstances",
+        "ec2:DescribeRegions"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+Note that it is possible to run a single global instance of this service that acts on hosts in all environments by using a global API key.
+Please note that if you take this approach, you need to manually set the CATTLE_* environment variables,
+as the "service account" approach described above creates single-environment keys only.
+
+#Configuration
+The following environment variables can be used to control the behaviour of this service:
+
+REAPER_INTERVAL_SECS - the interval in seconds between checking host status (default: 30).
+If set to "-1" the container will run in "one-shot" mode,
+in which it will check hosts once and then shut down.
+This is useful if you're running the container using an external scheduler such as Rancher cron.
+REAPER_DRY_RUN - If set to "true", this service will simply log what it would do without actually doing it (default: false)
+REAPER_INSTANCE_ID_LABEL_NAME - The name of the Rancher host label that holds the AWS instance ID. Defaults to aws.instance_id.
+REAPER_AVAILABILITY_ZONE_LABEL_NAME - The name of the Rancher host label that holds the AWS availability zone. Defaults to aws.availability_zone.
+##Developing
+Suggestions and pull requests welcome at the GitHub repo.
+
+To run locally in Docker, set the environment variables listed above and run:
+
+$ docker-compose up
+Or without docker (local Ruby 2.x installation required):
+
+$ bundle install
+$ bundle exec thin -R lib/config.ru start
```

Comment:
The formatting of this is completely broken, can you add formatting marks?

Reply:
Sorry, yep.